Source/Core/Request.js

```diff
@@ -0,0 +1,162 @@
+
+        var throttleByServer = defaultValue(options.throttleByServer, false);
+        var throttle = throttleByServer || defaultValue(options.throttle, false);
+
+        /**
+         * The URL to request.
+         *
+         * @type {String}
+         */
+        this.url = options.url;
+
+        /**
+         * The actual function that makes the request. The function takes no arguments and returns a promise for the requested data.
+         *
+         * @type {Function}
+         *
+         * @private
+         */
+        this.requestFunction = options.requestFunction;
+
+        /**
+         * Whether to throttle and prioritize the request. If false, the request will be sent immediately. If true, the
+         * request will be throttled and sent based on priority.
+         *
+         * @type {Boolean}
+         * @readonly
+         *
+         * @default false
+         */
+        this.throttle = throttle;
+
+        /**
+         * Whether to throttle the request by server. Browsers typically support about 6-8 parallel connections
+         * for HTTP/1 servers, and an unlimited amount of connections for HTTP/2 servers. Setting this value
+         * to <code>true</code> is preferable for requests going through HTTP/1 servers.
+         *
+         * @type {Boolean}
+         * @readonly
+         *
+         * @default false
+         */
+        this.throttleByServer = throttleByServer;
+
+        /**
+         * Type of request.
+         *
+         * @type {RequestType}
+         * @readonly
+         *
+         * @default RequestType.OTHER
+         */
+        this.type = defaultValue(options.type, RequestType.OTHER);
+
+        /**
+         * The distance from the camera, used to prioritize requests. This value may be edited continually to update
+         * the request's priority.
+         *
+         * @type {Number}
+         *
+         * @default 0.0
+         */
+        this.distance = defaultValue(options.distance, 0.0);
+
+        /**
+         * The screen space error, used to prioritize requests. This value may be edited continually to update
+         * the request's priority.
+         *
+         * @type {Number}
+         *
+         * @default 0.0
+         */
+        this.screenSpaceError = defaultValue(options.screenSpaceError, 0.0);
+
+        /**
+         * The request server, derived from the url.
+         *
+         * @type {String}
+         *
+         * @private
+         */
+        this.server = undefined;
+
+        /**
+         * The current state of the request.
+         *
+         * @type {RequestState}
+         *
+         * @private
+         */
+        this.state = RequestState.UNISSUED;
+
+        /**
+         * Reference to the underlying XMLHttpRequest so that it may be aborted in RequestScheduler.
+         *
+         * @type {Object}
+         *
+         * @private
+         */
+        this.xhr = undefined;
```

Comment:
> why not just issue the abort in the cancel call below?

I try to keep most of the request handling in `RequestScheduler.update`.

> I'm also a little concerned with leaky abstraction again here. Since we want more robust Node support long term, letting the xhr instance "escape" the loadWithXhr call may cause headaches down the road. Not a show stopper right now, but something we may want to consider looking into sooner rather than later post-merge.

Hm.. there could be a `cancelRequest` callback instead of storing the `xhr` here.